sdk/eventhub/event-hubs/test/misc.spec.ts

```diff
@@ -171,7 +180,7 @@ describe("Misc tests", function(): void {
       await breceiver.close();
       debug("received message: ", data);
       should.exist(data);
-      data.length.should.equal(5);
+      should.equal(data.length > 0, true);
```

Comment:
The change here is not just in the use of `should`, but what we are testing as well. Reason?

Reply:
Oops, yes - this got added in while troubleshooting for #3471.

Precisely the reason why I wanted to decouple things and finish this PR before looking into something that depends on it. 
cc @AlexGhiondea

Reply:
> Precisely the reason why I wanted to decouple things and finish this PR before looking into something that depends on it.

You can create a new branch from this branch to decouple the test failure investigations